StortSpelProjekt/Engine/src/Renderer/QuadManager.cpp

```diff
@@ -0,0 +1,182 @@
+#include "../Renderer/GPUMemory/ShaderResourceView.h"
+#include "../Renderer/Mesh.h"
+#include "../Renderer/Renderer.h"
+
+#include "../Misc/Window.h"
+#include "../Misc/AssetLoader.h"
+
+#include "../Events/EventBus.h"
+
+QuadManager::QuadManager()
+{
+}
+
+QuadManager::~QuadManager()
+{
+	EventBus::GetInstance().Unsubscribe(this, &QuadManager::pressed);
+
+	if (m_pQuad != nullptr)
+	{
+		delete m_pQuad;
+	}
+}
+
+void QuadManager::CreateQuad(float2 pos, float2 size, bool clickable, std::wstring texturePath)
+{
+	// Loof if we are going to overwrite the current quad
+	if (m_pQuad != nullptr)
+	{
+		delete m_pQuad;
+		m_pQuad = nullptr;
+		m_Clickable = false;
+
+		if (m_pQuadTexture != nullptr)
+		{
+			delete m_pQuadTexture;
+			m_pQuadTexture = nullptr;
+		}
+	}
+
+	if (m_pQuadTexture == nullptr && texturePath != L"NONE")
+	{
+		AssetLoader* al = AssetLoader::Get();
+		m_pQuadTexture = al->LoadTexture2D(texturePath);
+	}
+
+	m_Clickable = clickable;
+
+	float x = (pos.x * 2.0f) - 1.0f;
+	float y = ((1.0f - pos.y) * 2.0f) - 1.0f;
+	size.x = ((pos.x + size.x) * 2.0f) - 1.0f;
+	size.y = ((1.0f - (pos.y + size.y)) * 2.0f) - 1.0f;
+
+	std::vector<Vertex> m_Vertices = {};
+
+	DirectX::XMFLOAT3 normal = DirectX::XMFLOAT3{ 1.0, 1.0, 0.0 };
+	DirectX::XMFLOAT3 tangent = DirectX::XMFLOAT3{ 0.0, 0.0, 0.0 };
+
+	Vertex vertex = {};
+	vertex.pos = DirectX::XMFLOAT3{ x, y, 0.0 };
+	m_Positions["upper_left"] = float2{ vertex.pos.x, vertex.pos.y };
+	vertex.uv = DirectX::XMFLOAT2{ 0.0, 0.0 };
+	vertex.normal = normal;
+	vertex.tangent = tangent;
+	m_Vertices.push_back(vertex);
+
+	vertex.pos = DirectX::XMFLOAT3{ x, size.y, 0.0 };
+	m_Positions["lower_left"] = float2{ vertex.pos.x, vertex.pos.y };
+	vertex.uv = DirectX::XMFLOAT2{ 0.0, 1.0 };
+	m_Vertices.push_back(vertex);
+
+	vertex.pos = DirectX::XMFLOAT3{ size.x, y, 0.0 };
+	m_Positions["upper_right"] = float2{ vertex.pos.x, vertex.pos.y };
+	vertex.uv = DirectX::XMFLOAT2{ 1.0, 0.0 };
+	m_Vertices.push_back(vertex);
+
+	vertex.pos = DirectX::XMFLOAT3{ size.x, size.y, 0.0 };
+	m_Positions["lower_right"] = float2{ vertex.pos.x, vertex.pos.y };
+	vertex.uv = DirectX::XMFLOAT2{ 1.0, 1.0 };
+	m_Vertices.push_back(vertex);
+
+	std::vector<unsigned int> indices = { 0, 1, 2, 1, 2, 3 };
+
+	Renderer* renderer = &Renderer::GetInstance();
+
+	m_pQuad = new Mesh(renderer->m_pDevice5, &m_Vertices, &indices, renderer->m_DescriptorHeaps[DESCRIPTOR_HEAP_TYPE::CBV_UAV_SRV]);
+
+	m_pQuad->m_pUploadResourceVertices = new Resource(renderer->m_pDevice5, m_pQuad->GetSizeOfVertices(), RESOURCE_TYPE::UPLOAD, L"Vertex_UPLOAD_RESOURCE");
+	m_pQuad->m_pDefaultResourceVertices = new Resource(renderer->m_pDevice5, m_pQuad->GetSizeOfVertices(), RESOURCE_TYPE::DEFAULT, L"Vertex_DEFAULT_RESOURCE");
+
+	// Vertices
+	const void* data = static_cast<const void*>(m_pQuad->m_Vertices.data());
+	Resource* uploadR = m_pQuad->m_pUploadResourceVertices;
+	Resource* defaultR = m_pQuad->m_pDefaultResourceVertices;
```

Comment:
Are you using these? these are supposed to be sent into a copyOnDemand->Submit function. But I can't see that you're submitting here? Also I dont think you should submit here.